src/engine/strat_engine/crypt/shared.rs

```diff
@@ -236,14 +255,16 @@ fn all_tang_configs_have_url_trust_info(
     } else if pin == "sss" {
         if let Some(obj) = clevis_config.as_object() {
             if let Some(obj) = obj.get("pins").and_then(|val| val.as_object()) {
-                obj.iter().try_fold(true, |b, (pin, config)| {
-                    Ok(
-                        b && all_tang_configs_have_url_trust_info(
-                            pin,
-                            config,
-                            recursion_limit - 1,
-                        )?,
-                    )
+                obj.iter().try_fold(sss.0.clone(), |acc, (pin, config)| {
```

Comment:
Can you explain why you opted for a reference here? My brief analysis of the ownership here leads me to believe that a move and an owned data type would be okay here.

Reply:
I can change sss.0 so that it is not a reference. I've added a commit for that. But you can't move sss into the closure, because it is used after the closure returns. So, AFAICT, if you add a move, you have to be able to clone sss.

Reply:
I think this gets back to my other comment: why do all of these need to be references? I feel like ownership may remove the need for cloning.

Reply:
```sss``` is recursively passed and then used on the return value. I don't believe that it can be moved into the closure for that reason.